editor/src/messages/tool/tool_messages/path_tool.rs

```diff
@@ -228,9 +265,148 @@ impl Fsm for PathToolFsmState {
 								responses.push_back(DocumentMessage::DeselectAllLayers.into());
 							}
 						}
+
 						PathToolFsmState::Ready
 					}
 				}
+				//Rotating
+				(
+					PathToolFsmState::Rotating,
+					PathToolMessage::PointerMove {
+						alt_mirror_angle: _,
+						shift_mirror_distance,
+					},
+				) => {
+					let path = tool_data.shape_editor.selected_layers_ref();
+					let viewspace = &mut document.document_legacy.generate_transform_relative_to_viewport(path[0]).ok().unwrap();
+					// let to = document.generate_transform_relative_to_viewport(path[0]);
+
+					let points = tool_data.shape_editor.selected_points(&document.document_legacy);
+
+					//make sure that the pivot is in viewspace - fixed with the transofmr_point
+					let mut count: usize = 0;
+					let pivot = points
+						.map(|point| {
+							count += 1;
+							viewspace.transform_point2(point.position)
+						})
+						.sum::<DVec2>() / count as f64;
+
+					//drag start is in pixels // center is in relative pos
+					let vector_from_mouse_start = pivot - tool_data.grs_mouse_start;
```

Comment:
This needs to be `pivot - tool_data.previous_mouse_position` to avoid the massive spinning you are getting.